src/marshmallow/fields.py

```diff
@@ -1629,6 +1631,41 @@ def __init__(self, *args, **kwargs):
         self.validators = [validator] + original_validators
 
 
+class IP(String):
+    """A IP address field."""
+
+    default_error_messages = {"invalid_ip": "Not a valid IP address."}
+
+    def _validated(
+        self, value
+    ) -> typing.Optional[typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address]]:
+        """Format the value or raise a :exc:`ValidationError` if an error occurs."""
+        if value is None:
+            return None
+        if isinstance(value, (ipaddress.IPv4Address, ipaddress.IPv6Address)):
```

Comment:
Currently, marshmallow does not accept deseralized form when deserializing. This is discussed in #1415.

Reply:
Dropping lines 1645-1646 will actually have no effect on the interface level since `ipaddress.ip_address` function accepts `ipaddress.IPv4Address` and `ipaddress.IPv6Address` instances, and returns their copies.

Are you suggesting to add explicit type checking and accept only `str` values?

Reply:
I see no need to explicitly exclude deserialized form. But that could be the reason I'm in favour of #1415. I didn't check but I'd bet other fields already accept deserialized data.

In any case, I see no need for this check. As you said, the behaviour will stay the same.

Reply:
In that case, the issue is resolved via 985f2ba